api/src/pcapi/local_providers/allocine/allocine_stocks.py

```diff
@@ -209,9 +216,30 @@ def fill_stock_attributes(self, allocine_stock: Stock):  # type: ignore [no-unty
             allocine_stock.quantity = self.quantity
 
         if "price" not in allocine_stock.fieldsUpdated:
-            allocine_stock.price = self.apply_allocine_price_rule(allocine_stock)
-
-    def apply_allocine_price_rule(self, allocine_stock: Stock) -> decimal.Decimal:
+            price = self.apply_allocine_price_rule(allocine_stock)
+            allocine_stock.price = price
+            if not allocine_stock.priceCategory:
+                if allocine_stock.offerId in self.priceCategoryByOffer:
+                    allocine_stock.priceCategory = self.priceCategoryByOffer[allocine_stock.offerId]
+                else:
+                    price_category = self.get_or_create_allocine_price_category(price, allocine_stock.offerId)
+                    allocine_stock.priceCategory = price_category
+                    self.priceCategoryByOffer[allocine_stock.offerId] = price_category
+            else:
+                allocine_stock.priceCategory.price = price
+
+    def get_or_create_allocine_price_category(
+        self, price: decimal.Decimal, offer_id: int
+    ) -> offers_models.PriceCategory:
+        price_category = offers_models.PriceCategory.query.filter_by(price=price, offerId=offer_id).one_or_none()
+        if not price_category:
+            if not self.label:
+                self.label = offers_api.get_or_create_label("Tarif unique", self.venue)
+                self.label.id = get_next_label_id_from_database()
+            return offers_models.PriceCategory(priceCategoryLabelId=self.label.id, price=price, offerId=offer_id)
+        return price_category
```

Comment:
(style) en inversant la condition `if` tu as moins de code indenté et le flux de logique devient un peu plus limpide 

```suggestion
        if price_category:
                return price_category

        if not self.label:
            self.label = offers_api.get_or_create_label("Tarif unique", self.venue)
            self.label.id = get_next_label_id_from_database()
        return offers_models.PriceCategory(priceCategoryLabelId=self.label.id, price=price, offerId=offer_id)
```